<launch>

  <arg name="NUM" default="0"/>
  <arg name="HAND" default="right"/>

  <!-- control frequency -->
  <arg name="HZ" default="200"/>

  <arg name="MAINTAIN" default="true" />

  <!-- optional name for the initial pose -->
  <arg name="POSE" default="" />

  <!-- time before the controller gives up (seconds) -->
  <arg name="TIME_LIMIT" default="4.0" />

  <!-- which rosparam namespace to read the predefined poses -->
  <arg name="NS_POSE" default="allegro_kdl/poses" />

  <arg name="TORQUE_TOPIC" default="position_torque"/>

  <!-- cartesian velocity limits, linear (meters/s) and angular (radians/s) -->
  <arg name="VELOCITY_LIMIT_LINEAR" default="0.03"/>
  <arg name="VELOCITY_LIMIT_ANGULAR" default="0.35"/>

  <!-- reload gains -->
  <rosparam file="$(find allegro_hand_kdl)/config/gains.yaml" command="load"
    ns="allegroHand_$(arg HAND)_$(arg NUM)"/>

  <!-- reload predefined poses -->
  <rosparam command="load" file="$(find allegro_hand_kdl)/config/cartesian_poses.yaml" ns="$(arg NS_POSE)" />

  <node name="cartesian_pose_server"
        type="cartesian_pose_server"
        pkg="allegro_hand_kdl"
        output="screen"
        launch-prefix-no="xterm -e gdb --args" >

    <param name="hz" value="$(arg HZ)"/>
    <param name="pose_ns" value="$(arg NS_POSE)" />
    <param name="maintain_pose" value="$(arg MAINTAIN)" />
    <param name="initial_pose" value="$(arg POSE)" />
    <param name="time_limit" value="$(arg TIME_LIMIT)" />
    <param name="velocity_limit_linear" value="$(arg VELOCITY_LIMIT_LINEAR)" />
    <param name="velocity_limit_angular" value="$(arg VELOCITY_LIMIT_ANGULAR)" />

    <!-- topic remappings -->
    <remap from="joint_states" to="allegroHand_$(arg NUM)/joint_states"/>
    <remap from="position_torque" to="$(arg TORQUE_TOPIC)"/>

  </node>

</launch>
